packages/kas/src/nodes.ts

```diff
@@ -98,25 +134,43 @@ class Expr {
     }
 
     // an abstraction for chainable, bottom-up recursion
-    recurse(method, ...passed) {
+    recurse(method: string, ...passed: any[]): this {
```

Comment:
I see a lot of `any`s, is that because we don't understand the types or because inheritance dictates what types will be used by different implementations?

Reply:
This method is highly dynamic which makes typing it correctly difficult. 😞  I think we may be able to make this better using method overloads, but I don't want to block these changes on that.  I'll add a TODO here to tackle this later.  The return type though at least gives us some amount of type safety.